Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs

```diff
@@ -169,35 +171,112 @@ public static AuthenticationBuilder AddCustomAuthentication(this IServiceCollect
                 .AddScheme<AuthenticationSchemeOptions, CustomAuthenticationHandler>(AuthenticationSchemes.CustomAuthentication, null);
         }
 
+        /// <summary>
+        /// Sets up the proxy configuration based on the addresses in <paramref name="userList"/>.
+        /// </summary>
+        /// <param name="config">The <see cref="NetworkConfiguration"/> containing the config settings.</param>
+        /// <param name="userList">The string array to parse.</param>
+        /// <param name="options">The <see cref="ForwardedHeadersOptions"/> instance.</param>
+        /// <param name="optionType">1 to assign valid IP addresses to KnownProxies, 2 to assign to KnownNetworks.</param>
+        public static void ParseList(NetworkConfiguration config, string[] userList, ForwardedHeadersOptions options, int optionType)
+        {
+            for (var i = 0; i < userList.Length; i++)
+            {
+                if (IPNetAddress.TryParse(userList[i], out var addr))
+                {
+                    if ((!config.EnableIPV4 && addr.AddressFamily == AddressFamily.InterNetwork)
+                         || (!config.EnableIPV6 && addr.AddressFamily == AddressFamily.InterNetworkV6))
+                    {
+                        continue;
+                    }
+
+                    if (config.EnableIPV6 && config.EnableIPV4 && addr.AddressFamily == AddressFamily.InterNetwork)
+                    {
+                        // If the server is using dual-mode sockets, IPv4 addresses are supplied in an IPv6 format.
+                        // https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-5.0 .
+                        addr.Address = addr.Address.MapToIPv6();
+                    }
+
+                    if (optionType == 1)
```

Comment:
If IPNetAddress.TryParse is true then it'll always be an IPNetwork. You simply don't even need that optionType. I don't understand why you didn't accept my pr in the first place, which worked as intended, and instead implemented something that is overly complex, contains a ton of duplicated code and doesn't work.

Reply:
I'm a bit out of the loop lately, so I'm not entirely familiar with the network code anymore, but I do agree that it should be possible to simplify this bit. I do, however, _not_ appreciate your tone. Please keep it civil.

Reply:
Sorry, but my good tone has left the room after testing something the third time, which I had working three days ago... Simply ignoring my pr with working and tested code is also not good collaboration.

Reply:
Your PR is on his fork, so that's between you and him and better suited for a private chat. You're also free to submit your PR to this repo. Whichever PR is deemed the better solution will always get merged.

Reply:
1) The IP6/IP4 checking. (lines 188, 214)
The system now support dual IPv6/IPv4 network stacks. Not everyone has both IP4/6 enabled. Some have a system whereby the OS has IP6 enabled, but the user doesn't want IP6 enabled on JF. Some might have the reverse. 
Under this context, you have to be careful what values you add, otherwise you could be enabling the the wrong IP functionality and break something for someone else. Code can easily become complicated.

2) IP4 mapping. (lines 194, 221)
If the IP6 stack is enabled, the documentation clearly states that the addresses need to be in IP6 format regardless. This means any IP4 addresses need to be converted. Adding a parsed value without checking then introduces the possibility of future support issues being logged.

3) Correct, you could get rid of the optionType, but a good part of this code would then need to be duplicated. 
This method populates both KnownProxies and KnownNetwork depending upon the value of optionType.
Not the best method, granted. But seen as KnownProxies contains IP addresses, and KnownNetworks IP networks some knowledge of which property is being worked upon is required.

4) In addition, additional complexity is added with the repeating code after IPNetAddress and IPHost. This enables the ability that both IP addresses **OR** DNS names can be used with either of these settings.

5) **This is a PR**. It is not part of the main code, and thus is open for change/modification. 
It also might not make the next version of JF, or even be accepted at all. 
This is how the process works. Any changes/mods are made on this PR by submitting changes, and are reviewed by the community. 
You can spend weeks on a PR, only for it to be torn apart, or rejected. That is how the system works. Everything is written for the community, and not an individual.

6) Not everyone has the same configuration/setup so, at times we write code which we need the community to test, and retest and retest, and maybe even retest again.
If they don't, or are unwilling - then that functionality just won't be implemented.

7) We write code for the enjoyment of it...

Reply:
@ggogel 

Duplicate the settings that you have in the _network.xml_ under <KnownProxies> to the new key <KnownNetworks>.

```
<KnownProxies>
<string>192.168.179.0</string>
<string>10.0.0.0/8</string>
<KnownProxies>

<KnownNetworks>
<string>192.168.179.0</string>
<string>10.0.0.0/8</string>
<KnownNetworks>
```

Reply:
3. I can't comprehend why you split up the known proxy config into two lists in the first place. Sure this is how it's represented in .NET but take the "LAN networks" setting for instance, where you also can enter both IPs and netmasks in the same field. Without that split you wouldn't need the optionType, nor a lot of the duplicate code, which is basically there to cover cases when a user enters an IP into the network field or vice versa.
4. What happens when a host has multiple IPs (DNS round-robin), which is pretty much the case when you have multiple replicas running of the same service (here a reverse proxy).
6. The current implementation on the master branch is just broken. As soon as you enter something in the known proxies field, nothing is trusted anymore even when you use only one proxy. This will cause issues after release, I guarantee you that. So known proxies should be removed entirely or a fix should be found.

Reply:
Have taken your idea and have removed the KnownNetworks property. If the address in KnownProxies is a network address is it added to KnownNetworks. If a single, then known Proxies.

As for dns - it uses all addresses returned by dotnet after resolving the host name.

Reply:
I've tested the build now with the currently latest commit https://github.com/jellyfin/jellyfin/pull/4978/commits/126c64e1d90c2dd356065a22e3675a6988dcd6c4 and it works: 
```
[12:26:13] [DBG] [19] Emby.Server.Implementations.HttpServer.WebSocketConnection: WS 192.168.179.54 received message:
```
 Though the debug log on startup outputs the type of the list and not the content.

Reply:
Fixing the type thing now - along with the tests as one fails on the Mac.

Reply:
Can you confirm that this version still works? I need to remove the debug logging for the end product,

Reply:
I've just tested the dns functionality. I've entered the hostname of the two reverse proxy replicas and it seems to resolve their ip. This is actually a great addition for k8s and swarm, because labels / hostnames is the common way to identify services there.

Reply:
I can also confirm the latest version https://github.com/jellyfin/jellyfin/pull/4978/commits/2979c8dd377bf0948be00082b97416b38c21de56 is still working.